client/my-sites/plans/current-plan/index.jsx

```diff
@@ -154,6 +155,40 @@ class CurrentPlan extends Component {
 		return <PaidPlanThankYou />;
 	}
 
+	renderEcommerceTrialPage() {
```

Comment:
This implementation effectively removes a _lot_ of logic around prompts and Tracks events. Might it be cleaner to have a conditional that injects our custom content?

@verofasulo, this may also be an area where we need your input, because there are multiple notices and warnings that we currently show on the My Plan page, and we need to work out how we should handle//inject them in the new layout.

Reply:
In terms of the implementation, @nelsonec87, it may be worth pulling some of the shared code like the `<Query*>` components and the `TrackComponentView` out into helper functions and then inject them in both locations or code paths. Right now there is a _lot_ of duplication that feels like it is really likely to introduce bugs in the long term.